File: ios/USTAQuery/Models/Ranking.swift
import Foundation

struct Ranking: Codable, Identifiable, Hashable {
    let id: Int
    let playerUaid: String
    let playerFirstName: String
    let playerLastName: String
    let catalogId: String
    let displayLabel: String?
    let playerType: String?
    let ageRestriction: String?
    let ageRestrictionModifier: String?
    let rankListGender: String?
    let listType: String?
    let matchFormat: String?
    let matchFormatType: String?
    let familyCategory: String?
    let nationalRank: Int?
    let sectionRank: Int?
    let districtRank: Int?
    let points: Int?
    let singlesPoints: Int?
    let doublesPoints: Int?
    let bonusPoints: Int?
    let wins: Int?
    let losses: Int?
    let trendDirection: String?
    let publishDate: String?
    let section: String?
    let district: String?
    let state: String?
}

File: android/app/src/main/java/com/usta/query/data/model/Ranking.kt
package com.usta.query.data.model

data class Ranking(
    val id: Int,
    val playerUaid: String,
    val playerFirstName: String,
    val playerLastName: String,
    val catalogId: String,
    val displayLabel: String? = null,
    val playerType: String? = null,
    val ageRestriction: String,
    val ageRestrictionModifier: String? = null,
    val rankListGender: String,
    val listType: String,
    val matchFormat: String? = null,
    val matchFormatType: String? = null,
    val familyCategory: String? = null,
    val nationalRank: Int? = null,
    val sectionRank: Int? = null,
    val districtRank: Int? = null,
    val points: Int? = null,
    val singlesPoints: Int? = null,
    val doublesPoints: Int? = null,
    val bonusPoints: Int? = null,
    val wins: Int? = null,
    val losses: Int? = null,
    val trendDirection: String? = null,
    val publishDate: String? = null,
    val section: String? = null,
    val district: String? = null,
    val state: String? = null
)

data class RankingHistory(
    val playerUaid: String,
    val catalogId: String,
    val displayLabel: String? = null,
    val dataPoints: List<Ranking> = emptyList()
)
